kino-nextjs/app/components/movies/BookingSeat.tsx

```diff
@@ -0,0 +1,120 @@
+"use client";
+import React, { useState } from "react";
+import { useTickets } from "../../contexts/TicketContext";
+import { createBooking } from "../../lib/actions";
+import { screening, booking } from "../../lib/types";
+
+interface BookingSeatProps {
+  screening: screening;
+}
+
+const BookingSeat: React.FC<BookingSeatProps> = ({ screening }) => {
+  const [seats, setSeats] = useState<(0 | 1 | 2)[][]>(screening.Seats);
+  const [totalPrice, setTotalPrice] = useState<number>(0);
+  const { tickets } = useTickets();
+  const initialSeats = screening.Seats;
+
+  const getTotalSelectedTickets = () => {
+    return tickets.adult + tickets.senior + tickets.child;
+  };
+
+  const getSelectedSeatsCount = () => {
+    return seats.flat().filter((seat) => seat === 2).length;
+  };
+
+  const toggleSeat = (rowIndex: number, seatIndex: number) => {
+    setSeats((prevSeats) =>
+      prevSeats.map((row, rIndex) =>
+        rIndex === rowIndex
+          ? row.map((seat, sIndex) =>
+              sIndex === seatIndex
+                ? seat === 0
+                  ? getSelectedSeatsCount() < getTotalSelectedTickets()
+                    ? 2
+                    : seat
+                  : seat === 2
+                  ? 0
+                  : seat
+                : seat

```

Comment:
Kind of hard (for me at least) to understand what's going on here, but seems to be working :)